### DISCUSSION TITLE
Forest NV24 :auto_rickshaw: support

### DISCUSSION THREAD

**Discussion by LesnyRumcajs** [ChainSafe/forest#4883] [Announcements 📢]
### NV24 TukTuk :auto_rickshaw: upgrade

Forest joins Lotus and Venus in supporting the [NV24 upgrade](https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-10807265). See the post for details regarding the new features brought to the network.

To keep Forest nodes running, the operators **must** upgrade Forest to supporting versions, as outlined in the timeline.

### :calendar: Timeline 

```mermaid
timeline
    title NV24 Forest timeline
    ~October 17th : Forest release supporting NV24 for calibnet
    October 23rd : Calibration network upgrade
    ~November 10th : Forest release, supporting NV24 for mainnet
    November 20th : Mainnet network upgrade
```

The timeline is subject to changes, and there will likely be non-mandatory releases. Depending on external factors, the mainnet release will likely arrive faster to give plenty of time for an update.

### :computer: Hardware requirements

The state migration for this upgrade is minimal, and shouldn't require significantly more resources than a regular node usage. Expected migration durations:
- calibnet: below 1s,
- mainnet: below 30s.

An alternative to the automatic state migration (and also when you are not a Forest user :crying_cat_face: ) is downloading a snapshot from the ChainSafe snapshot service sometime after the upgrade (30m - 2h). Endpoints below:
* https://forest-archive.chainsafe.dev/latest/mainnet/
* https://forest-archive.chainsafe.dev/latest/calibnet/

### :scroll:  Read more about network upgrades

If you haven't already, we invite you to familiarize yourself with the network upgrade process with [this page](https://forest-docs-v2.pages.dev/knowledge_base/network_upgrades_state_migrations) in the Forest documentation.

**Comment by LesnyRumcajs** [comment]
[Forest v0.21.0 "Songthaew"](https://github.com/ChainSafe/forest/releases/tag/v0.21.0) released! :evergreen_tree: 

Please upgrade your calibnet nodes before epoch 2078794, which is `2024-10-23T13:30:00Z`, to enjoy full _TukTuk_ :auto_rickshaw: support!

* :octocat: Release sources and binaries - https://github.com/ChainSafe/forest/releases/tag/v0.21.0
* :whale: Docker image - https://github.com/ChainSafe/forest/pkgs/container/forest/290189977?tag=v0.21.0-fat
* :package: crates.io - https://crates.io/crates/forest-filecoin

What is [_Songthaew_](https://en.wikipedia.org/wiki/Songthaew)? It's kind of a Tuk Tuk, but larger!

![image](https://github.com/user-attachments/assets/7cc41f7f-24fd-4375-bc10-fb9e4716ff96)

**Comment by hanabi1224** [comment]
[Forest v0.21.1 "Songthaew Plus"](https://github.com/ChainSafe/forest/releases/tag/v0.21.1) released! :evergreen_tree: 

Please upgrade your calibnet nodes to get F3 running!

* :octocat: Release sources and binaries - https://github.com/ChainSafe/forest/releases/tag/v0.21.1
* :whale: Docker image - https://github.com/ChainSafe/forest/pkgs/container/forest/290189977?tag=v0.21.1-fat
* :package: crates.io - https://crates.io/crates/forest-filecoin

What is Fast Finality in Filecoin (F3)? See https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0086.md for details.

**Comment by LesnyRumcajs** [comment]
[Forest v0.22.0 "Pad Thai"](https://github.com/ChainSafe/forest/releases/tag/v0.22.0) released! 🌲 

This is a mandatory release for the mainnet nodes, preparing them for the NV24 _Tuk Tuk_ 🛺 upgrade at epoch 4_461_240 corresponding to `Wed 20 Nov 23:00:00 UTC 2024`. Please upgrade your node before that date to ensure a smooth transition into NV24.

Please not that F3 is no longer planned for NV24; see [this post](https://github.com/filecoin-project/core-devs/discussions/150#discussioncomment-11164504) for more details.

* :octocat: Release sources and binaries - https://github.com/ChainSafe/forest/releases/tag/v0.22.0
* :whale: Docker image - https://github.com/ChainSafe/forest/pkgs/container/forest/302402004?tag=v0.22.0-fat
* :package: crates.io - https://crates.io/crates/forest-filecoin
